include/osquery/tables.h

```diff
@@ -759,6 +759,33 @@ class TablePlugin : public Plugin {
     return QueryData();
   }
 
+  /// Callback for DELETE statements
+  virtual QueryData delete_(QueryContext& context,
+                            const PluginRequest& request) {
+    static_cast<void>(context);
```

Comment:
May I ask you to use in such cases `boost::ignore_unused` as more explicit way to switch off this warning. [ref](https://www.boost.org/doc/libs/1_67_0/libs/core/doc/html/core/ignore_unused.html)